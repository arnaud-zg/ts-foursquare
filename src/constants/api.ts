export enum EApiDefaultParameters {
  ORIGIN = 'https://api.foursquare.com',
  VERSION = '20191028',
}

export enum EApiPathnames {
  LISTS = '/v2/lists/:listId',
  PHOTOS_DETAILS = '/v2/photos/:photoId',
  VENUES_CATEGORIES = '/v2/venues/categories',
  VENUES_EXPLORE = '/v2/venues/explore',
  VENUES_LIKES = '/v2/venues/:venueId/likes',
  VENUES_LISTED = '/v2/venues/:venueId/listed',
  VENUES_NEXT_VENUES = '/v2/venues/:venueId/nextvenues',
  VENUES_SEARCH = '/v2/venues/search',
  VENUES_SIMILAR = '/v2/venues/:venueId/similar',
  VENUES_SUGGEST_COMPLETION = '/v2/venues/suggestcompletion',
  VENUES_TRENDING = '/v2/venues/trending',
}
