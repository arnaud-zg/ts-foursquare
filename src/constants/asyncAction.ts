import { getType } from 'typesafe-actions'
import {
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
} from '../DEPRECATED_actions'

interface IAsyncActionNameMapping {
  [actionType: string]: string
}

export const ASYNC_ACTION_NAME_MAPPING: IAsyncActionNameMapping = {
  // getVenuesCategoriesAsync
  [getType(getVenuesCategoriesAsync.request)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.success)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.failure)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.cancel)]: 'getVenuesCategoriesAsync',
  // getVenuesExploreAsync
  [getType(getVenuesExploreAsync.request)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.success)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.failure)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.cancel)]: 'getVenuesExploreAsync',
  // getVenuesLikesAsync
  [getType(getVenuesLikesAsync.request)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.success)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.failure)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.cancel)]: 'getVenuesLikesAsync',
  // getVenuesListedAsync
  [getType(getVenuesListedAsync.request)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.success)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.failure)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.cancel)]: 'getVenuesListedAsync',
  // getVenuesNextVenuesAsync
  [getType(getVenuesNextVenuesAsync.request)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.success)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.failure)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.cancel)]: 'getVenuesNextVenuesAsync',
  // getVenuesSearchAsync
  [getType(getVenuesSearchAsync.request)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.success)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.failure)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.cancel)]: 'getVenuesSearchAsync',
  // getVenuesSimilarAsync
  [getType(getVenuesSimilarAsync.request)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.success)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.failure)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.cancel)]: 'getVenuesSimilarAsync',
  // getVenuesSuggestCompletionAsync
  [getType(
    getVenuesSuggestCompletionAsync.request
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.success
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.failure
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.cancel
  )]: 'getVenuesSuggestCompletionAsync',
  // getVenuesTrendingAsync
  [getType(getVenuesTrendingAsync.request)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.success)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.failure)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.cancel)]: 'getVenuesTrendingAsync',
}
