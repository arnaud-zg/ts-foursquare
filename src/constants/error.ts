export enum ECommonError {
  RESPONSE_NOT_VALID = 'Response is not valid',
}
