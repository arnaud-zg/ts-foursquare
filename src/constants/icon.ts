export enum EIconSize {
  S = '32',
  M = '44',
  L = '64',
  XL = '88',
}
