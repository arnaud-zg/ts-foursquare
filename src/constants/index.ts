export const PROJECT_NAME = 'ts-foursquare'
