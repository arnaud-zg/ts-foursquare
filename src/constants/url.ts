export enum EUrlSeparator {
  LOCATION_SEARCH_SEPARATOR = '?',
  QUERY_STRING_SEPARATOR = '&',
}
