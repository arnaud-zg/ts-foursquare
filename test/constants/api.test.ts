import { EApiDefaultParameters, EApiPathnames } from '../../src/constants/api'

describe('EApiDefaultParameters', () => {
  it('should make a snapshot of EApiDefaultParameters', () => {
    expect(EApiDefaultParameters).toMatchSnapshot()
  })

  it('should make a snapshot of EApiPathnames', () => {
    expect(EApiPathnames).toMatchSnapshot()
  })
})
