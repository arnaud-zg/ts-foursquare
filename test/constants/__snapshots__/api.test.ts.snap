// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constants/api/EApiDefaultParameters should make a snapshot of EApiDefaultParameters 1`] = `
Object {
  "ORIGIN": "https://api.foursquare.com",
  "VERSION": "20191028",
}
`;

exports[`constants/api/EApiDefaultParameters should make a snapshot of EApiPathnames 1`] = `
Object {
  "LISTS": "/v2/lists/:listId",
  "PHOTOS_DETAILS": "/v2/photos/:photoId",
  "VENUES_CATEGORIES": "/v2/venues/categories",
  "VENUES_EXPLORE": "/v2/venues/explore",
  "VENUES_LIKES": "/v2/venues/:venueId/likes",
  "VENUES_LISTED": "/v2/venues/:venueId/listed",
  "VENUES_NEXT_VENUES": "/v2/venues/:venueId/nextvenues",
  "VENUES_SEARCH": "/v2/venues/search",
  "VENUES_SIMILAR": "/v2/venues/:venueId/similar",
  "VENUES_SUGGEST_COMPLETION": "/v2/venues/suggestcompletion",
  "VENUES_TRENDING": "/v2/venues/trending",
}
`;
